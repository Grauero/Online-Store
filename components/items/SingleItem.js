import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import Head from 'next/head';

import ErrorMessage from '../error/ErrorMessage';
import SingleItemStyles from '../styles/SingleItemStyles';
import { SINGLE_ITEM_QUERY } from '../../queries/items';

const SingleItem = ({ id }) => (
  <Query query={SINGLE_ITEM_QUERY} variables={{ id }}>
    {({ error, loading, data: { item } }) => {
      if (error) {
        return <ErrorMessage error={error} />;
      }
      if (loading) {
        return <p>Loading...</p>;
      }
      if (!item) {
        return <p>No Item Found for {id}</p>;
      }

      return (
        <SingleItemStyles>
          <Head>
            <title>Online Store | {item.title}</title>
          </Head>
          <img src={item.largeImage} alt={item.title} />
          <div className="details">
            <h2>Viewing {item.title}</h2>
            <p>{item.description}</p>
          </div>
        </SingleItemStyles>
      );
    }}
  </Query>
);

SingleItem.propTypes = {
  id: PropTypes.string.isRequired
};

export default SingleItem;
