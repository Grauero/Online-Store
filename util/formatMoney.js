export default function (amount) {
  if (typeof amount !== 'number' || amount < 0) {
    return 'Incorrect format, provide a positive number value as argument';
  }

  const options = {
    style: 'currency',
    currency: 'USD',
    minimumFractionDigits: 2
  };

  if (amount % 100 === 0) {
    options.minimumFractionDigits = 0;
  }
  const formatter = new Intl.NumberFormat('en-US', options);

  return formatter.format(amount / 100);
}
