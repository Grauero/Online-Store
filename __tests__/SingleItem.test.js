import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';

import { fakeItem } from '../util/testUtils';
import SingleItem, { SINGLE_ITEM_QUERY } from '../components/items/SingleItem';

it('renders with proper data', async () => {
  const mocks = [
    {
      // when use request with this query and variable
      request: { query: SINGLE_ITEM_QUERY, variables: { id: '123' } },
      // return test data (mocked data)
      result: {
        data: {
          item: fakeItem()
        }
      }
    }
  ];

  const wrapper = mount(
    <MockedProvider mocks={mocks}>
      <SingleItem id="123" />
    </MockedProvider>
  );

  expect(wrapper.text()).toContain('Loading...');
  await wait();
  wrapper.update();

  expect(toJSON(wrapper.find('h2'))).toMatchSnapshot();
  expect(toJSON(wrapper.find('img'))).toMatchSnapshot();
  expect(toJSON(wrapper.find('p'))).toMatchSnapshot();
});
