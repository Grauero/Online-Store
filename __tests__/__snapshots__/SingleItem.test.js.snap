// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleItem/> Errors with a not found item 1`] = `
<p
  data-test="graphql-error"
>
  Items Not Found!
</p>
`;

exports[`<SingleItem/> renders with proper data 1`] = `
<h2>
  Viewing 
  title
</h2>
`;

exports[`<SingleItem/> renders with proper data 2`] = `
<img
  alt="title"
  src="largeImage"
/>
`;

exports[`<SingleItem/> renders with proper data 3`] = `
<p>
  description
</p>
`;
